misc/python/materialize/mzbuild.py

```diff
@@ -605,31 +614,27 @@ def build(self) -> None:
         ]
         spawn.runv(cmd, stdin=f, stdout=sys.stderr.buffer)
 
-    def acquire(self) -> None:
-        """Download or build the image if it does not exist locally."""
-        if self.acquired:
-            return
-
-        ui.header(f"Acquiring {self.spec()}")
-        try:
-            spawn.runv(
-                ["docker", "pull", self.spec()],
-                stdout=sys.stderr.buffer,
-            )
-        except subprocess.CalledProcessError:
-            self.build()
-        self.acquired = True
-
-    def ensure(self) -> None:
-        """Ensure the image exists on Docker Hub if it is publishable.
+    def acquire(self) -> bool:
+        """Download the image if it does not exist locally. Returns whether it was found."""
+        if not self.acquired:
+            ui.header(f"Acquiring {self.spec()}")
+            try:
+                spawn.runv(
+                    ["docker", "pull", self.spec()],
+                    stdout=sys.stderr.buffer,
+                )
+            except subprocess.CalledProcessError:
+                pass
+            else:
+                self.acquired = True
+        return self.acquired
 
-        The image is pushed using its spec as its tag.
-        """
+    def ensure(self) -> bool:
+        """Ensure the image exists on Docker Hub if it is publishable. Returns whether it exists."""
```

Comment:
```suggestion
    def is_published_if_necessary(self) -> bool:
        """Report whether the image exists on Docker Hub if it is publishable."""
```

(Same rationale as the last comment.)